kernel/src/path.rs

```diff
@@ -199,6 +199,88 @@ mod tests {
     }
 
     #[test]
+    fn test_invalid_version_and_part_lengths() {
+        let table_log_dir = table_log_dir_url();
+
+        // Test cases for VERSION_LEN (20 digits)
+        let test_cases = vec![
+            // Error expected: Version number is 19 digits, but should be 20
+            ("000000000000000006.json", "short version number"),
+            // Error expected: Version number is 21 digits, but should be 20
+            ("000000000000000000006.json", "long version number"),
+        ];
+
+        for (file_name, error_msg) in test_cases {
+            let log_path = table_log_dir.join(file_name).unwrap();
+            let result = ParsedLogPath::try_from(log_path);
+            assert!(result.is_err(), "Expected an error for {}", error_msg);
+            assert!(
+                matches!(result, Err(Error::InvalidLogPath(_))),
+                "Expected InvalidLogPath error for {}",
+                error_msg
+            );
+        }
+
+        // Test cases for MULTIPART_PART_LEN (10 digits)
+        let test_cases = vec![
+            // Error expected: Part number is 9 digits, but should be 10
+            (
+                "00000000000000000010.checkpoint.000000001.0000000002.parquet",
+                "short part number",
+            ),
+            // Error expected: Part number is 11 digits, but should be 10
+            (
+                "00000000000000000010.checkpoint.00000000001.0000000002.parquet",
+                "long part number",
+            ),
+            // Error expected: Total parts is 9 digits, but should be 10
+            (
+                "00000000000000000010.checkpoint.0000000001.000000002.parquet",
+                "short total parts",
+            ),
+            // Error expected: Total parts is 11 digits, but should be 10
+            (
+                "00000000000000000010.checkpoint.0000000001.00000000002.parquet",
+                "long total parts",
+            ),
+        ];
+
+        for (file_name, error_msg) in test_cases {
+            let log_path = table_log_dir.join(file_name).unwrap();
+            let result = ParsedLogPath::try_from(log_path);
+            assert!(result.is_err(), "Expected an error for {}", error_msg);
+            assert!(
+                matches!(result, Err(Error::InvalidLogPath(_))),
+                "Expected InvalidLogPath error for {}",
+                error_msg
+            );
+        }
+
+        // Test cases for UUID_PART_LEN (36 characters)
+        let test_cases = vec![
+            // Error expected: UUID is 35 characters, but should be 36
+            (
+                "00000000000000000010.checkpoint.3a0d65cd-4056-49b8-937b-95f9e3ee90.parquet",
```

Comment:
Technically already covered:
https://github.com/delta-incubator/delta-kernel-rs/blob/main/kernel/src/path.rs#L363-L366

... but this is nicer (checks the boundary case)

Reply:
It's off-by-one tho (only 34 chars):
```suggestion
            // Error expected: UUID is 35 characters, but should be 36
            (
                "00000000000000000010.checkpoint.3a0d65cd-4056-49b8-937b-95f9e3ee90e.parquet",
```

Reply:
Fixed char length and retained the case.